video_worker/tests/test_api_communicate.py

```diff
@@ -0,0 +1,237 @@
+            },
+        ),
+        # Success
+        (
+            {
+                'val_token': 'dummy-val-token',
+                'val_id': 'dummy-val-id',
+                'course_url': ['dummy-course-id']
+            },
+        )
+    )
+    @unpack
+    @patch('video_worker.api_communicate.requests.put')
+    @patch('video_worker.api_communicate.requests.post')
+    @patch('video_worker.api_communicate.requests.get')
+    @patch('video_worker.api_communicate.logger')
+    def test_send_val_data(self, data, mock_logger, mock_get, mock_post, mock_put):
+        """
+        Tests `send_val_data` method works correctly.
+        """
+        get_status_code = data.get('get_status_code', 200)
+        post_status_code = data.get('post_status_code', 201)
+        expected_error_message= data.get('error_message', '')
+        response_data = json.dumps({
+            'courses': [{
+                'dummy-course-id': 'dummy-video-image.jpeg'
+            }],
+            'encoded_videos': [{
+                'url': 'https://testurl.mp4',
+                'file_size' : 8499040,
+                'bitrate' : 131,
+                'profile' : 'override',
+            }]
+        })
+
+        mock_get.return_value = Mock(status_code=get_status_code, text=response_data, content=response_data)
+        mock_post.return_value = Mock(status_code=post_status_code, text=response_data, content=response_data)
+        mock_post.return_value = Mock(status_code=post_status_code, text=response_data, content=response_data)
+        communicate_api = self.setup_api(
+            VideoObject=self.Video,
+            val_video_status=self.val_video_status,
+            val_token = data.get('val_token', None),
+            val_id= data.get('val_id', None),
+            course_url=data.get('course_url', []),
+        )
+
+        response = communicate_api.send_val_data()
+
+        if not data.get('val_token', None):
+            self.assertFalse(response)
+
+        if expected_error_message:
+            mock_logger.error.assert_called_with(expected_error_message)
+
+    @data(
+        (False, 400, ''),
+        (True, 400, 'VEDA API Fail: Check VEDA API config'),
+        (True, 200, '')
+    )
+    @unpack
+    @patch('video_worker.api_communicate.requests.get')
+    @patch('video_worker.api_communicate.logger')
+    def test_determine_veda_pk(self, send_video_object, get_status_code, expected_error_message, mock_logger, mock_get):
+        """
+        Tests `determine_veda_pk` method works correctly.
+        """
+        response_data = json.dumps({
+            'results': [{
+                'id': 123
+            }]
+        })
+        mock_get.return_value = Mock(status_code=get_status_code, text=response_data, content=response_data)
+        communicate_api = self.setup_api(
+            VideoObject=self.Video if send_video_object else None
+        )
+
+        response = communicate_api.determine_veda_pk()
```

Comment:
I have left run, run_veda, run_val and __init__ methods because they are using these methods. It would be useless to write tests for them.